tests/test_json_schema_to_grammar.py

```diff
@@ -0,0 +1,325 @@
+import json
+
+import pytest
+from jsonschema import validate
+
+from guidance._grammar import GrammarFunction
+from guidance._json_schema_to_grammar import json_schema_to_grammar
+from guidance._parser import EarleyCommitParser, ParserException
+
+
+def to_compact_json(target: any) -> str:
+    # See 'Compact Encoding':
+    # https://docs.python.org/3/library/json.html
+    # Since this is ultimately about the generated
+    # output, we don't need to worry about pretty printing
+    # and whitespace
+    return json.dumps(target, separators=(",", ":"))
+
+
+def check_string_with_grammar(input_string: str, grammar: GrammarFunction):
+    parser = EarleyCommitParser(grammar)
+
+    print(f"Checking {input_string}")
+    for c in input_string:
+        print(f"Working on: {c}")
+        print(f"Valid next bytes: {parser.valid_next_bytes()}")
+        next_byte = bytes(c, encoding="utf8")
+        print(f"Consuming: {next_byte}")
+        parser.consume_byte(next_byte)
+
+
+@pytest.mark.parametrize(
+    "simple_json_string",
+    [
+        to_compact_json(x)
+        for x in [
+            "with_underscore",
+            "ALLCAPS",
+            "with a space",
+            "MiXeD cAsInG",
+            "with-hyphen",
+            "Mix case_underscore-hyphens",
+            "with a comma, in the string",
+            "A full stop.",
+            """How about
+            a
+            multiline string?""",
+            "A \t tab \t between \t words",
+            r"End with backslash \ ",
+            "Have a forward / slash",
+            "Include [the] odd {brace} and (parentheses)",
+            "Some more symbols: ; are useful!",
+        ]
+    ],
+)
+def test_string_schema(simple_json_string):
+    schema = """{ "type": "string" }"""
+
+    # First sanity check what we're setting up
+    schema_obj = json.loads(schema)
+    validate(instance=simple_json_string, schema=schema_obj)
+
+    # Now set up the actual conversion
+    grammar = json_schema_to_grammar(schema)
+    parser = EarleyCommitParser(grammar)
+
+    for c in simple_json_string:
+        print(f"Working on: {c}")
+        print(f"Valid next bytes: {parser.valid_next_bytes()}")
+        next_byte = bytes(c, encoding="utf8")
+        print(f"Consuming: {next_byte}")
+        parser.consume_byte(next_byte)
```

Comment:
Assume this can leverage the `check_string_with_grammar` function defined earlier right?

Reply:
This should all be changed to use `GrammarFunction.match()`